File: frontend/src/types/spec.ts
/** Mirrors backend/app/spec/models.py — JSON wire format. */

export interface SpecEntry {
  id: string;
  type: string;
  path: string;
  title: string;
  status: string;
  covers: string[];
  tags: string[];
  created: string;
  updated: string;
}

export interface Link {
  from: string;
  to: string;
  type: string;
}

export interface SpecSummary {
  id: string;
  type: string;
  path: string;
  status: string;
  title: string;
  tags: string[];
  covers: string[];
  created: string;
  updated: string;
}

export interface SpecDetail {
  id: string;
  type: string;
  path: string;
  status: string;
  title: string;
  tags: string[];
  content: string;
  links: Link[];
}

export interface DocumentEntry {
  path: string;
  title: string;
}

export interface SpecGraph {
  nodes: SpecEntry[];
  edges: Link[];
  documents: DocumentEntry[];
}

File: mobile/shared/src/commonMain/kotlin/dev/aiir/bonsai/data/model/Spec.kt
package dev.aiir.bonsai.data.model

import kotlinx.serialization.Serializable

@Serializable
data class Link(
    val from: String,
    val to: String,
    val type: String,
)

@Serializable
data class RegistryEntry(
    val id: String,
    val type: String,
    val path: String,
    val title: String = "",
    val status: String = "",
    val covers: List<String> = emptyList(),
    val tags: List<String> = emptyList(),
    val created: String = "",
    val updated: String = "",
)

@Serializable
data class SpecDetail(
    val id: String,
    val type: String,
    val path: String,
    val status: String = "",
    val title: String = "",
    val tags: List<String> = emptyList(),
    val content: String = "",
    val links: List<Link> = emptyList(),
)

@Serializable
data class SpecGraph(
    val nodes: List<RegistryEntry> = emptyList(),
    val edges: List<Link> = emptyList(),
)
